Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -10865,15 +10868,34 @@ def events_error_handler(res):
         ).format(xsiam_url=xsiam_url, headers=json.dumps(headers, indent=8), status_code=res.status_code, error=error)
 
         demisto.error(header_msg + api_call_info)
-        raise DemistoException(header_msg + error, DemistoException)
+        raise DemistoException(header_msg + error, DemistoException, res)
 
-    zipped_data = gzip.compress(data.encode('utf-8'))   # type: ignore[AttributeError,attr-defined]
+    zipped_data = gzip.compress(data.encode('utf-8'))  # type: ignore[AttributeError,attr-defined]
     client = BaseClient(base_url=xsiam_url)
-    res = client._http_request(method='POST', full_url=urljoin(xsiam_url, '/logs/v1/xsiam'), data=zipped_data,
-                               headers=headers,
-                               error_handler=events_error_handler)
-    if res.get('error').lower() != 'false':
-        raise DemistoException(header_msg + res.get('error'))
+
+    # retry mechanism in case there is a rate limit (429) from xsiam.
+    for attempt_num in range(1, num_of_attempts + 1):
+        demisto.debug('Sending events into xsiam: attempt number {attempt_num}'.format(attempt_num=attempt_num))
+        ok_codes = (200, 429) if attempt_num < num_of_attempts else ()
```

Comment:
will sending in an empty tuple work like you expect? Should probably just put 200 explicitly

Reply:
yes its working, cause by default we don't send any tuple

Reply:
what do you mean? we will send in ()

Reply:
changed to `None` cause that is the default when not sending anything to the `ok_codes`, when you don't send anything to `ok_codes` it will fallback to the `_is_status_code_valid` and will go into the `response.ok` function which is:

```
@property
    def ok(self):
        """Returns True if :attr:`status_code` is less than 400, False if not.

        This attribute checks if the status code of the response is between
        400 and 600 to see if there was a client error or a server error. If
        the status code is between 200 and 400, this will return True. This
        is **not** a check to see if the response code is ``200 OK``.
        """
        try:
            self.raise_for_status()
        except HTTPError:
            return False
        return True
```